balloon.py

```diff
@@ -178,7 +181,7 @@ def balloon_m(password, salt, space_cost, time_cost, parallel_cost, delta=3) ->
             parallel_salt = b"" + salt.encode("utf-8") + (p + 1).to_bytes(8, "little")
             futures.append(
                 executor.submit(
-                    balloon, password, parallel_salt, space_cost, time_cost, delta=delta
+                    balloon, password, parallel_salt, space_cost, time_cost, delta=delta  # type: ignore
```

Comment:
**balloon()** accepts the `salt` argument as a str. However, it is also capable of accepting int, str and bytes, because **hash_func()** handles all 3 types. The type error occurs because `parallel_salt` is passed in as bytes. We could alternatively set `salt` to accept str|bytes.

Reply:
I really know little about Python, I come from a very type-safe programming world, so I definitely prefer `str | bytes`, let me know if you think this is overkill or really not Python-like.